scipy/stats/_continuous_distns.py

```diff
@@ -6942,6 +6942,116 @@ def _stats(self, a):
     def _entropy(self, a):
         return 1 - 1.0/a - np.log(a)
 
+    def fit(self, data, *args, **kwds):
+        data, fshape, floc, fscale = _check_fit_input_parameters(self, data,
+                                                                 args, kwds)
+        # ensure that any fixed parameters don't violate constraints of the
+        # distribution before continuing. The support of the distribution
+        # is `0 < (x - loc)/scale < 1`.
+        if floc is not None:
+            if not data.min() >= floc:
+                raise FitDataError('powerlaw', 0, 1)
+            if fscale is not None and not data.max() <= floc + fscale:
+                raise FitDataError('powerlaw', 0, 1)
+
+        if fscale is not None and fscale <= 0:
+            raise Exception("Negative or zero `fscale` is outside the range "
+                            "allowed by the distribution.")
+
```

Comment:
Could you write a comment up here preparing the reader for the logic that follows? E.g. what are the different cases and (roughly) how is each treated? Maybe number the different branches and include markers in the code that follows? I'm just trying to think of ways to make the logic easier to verify. Maybe a condensed version of the word document I wrote for you without specific formulas or detailed strategies, but just a breakdown of the different cases? (I know you've been careful to include comments, and those are good. Yet without referring to that document, I'm still getting lost!)

Reply:
Sure.